test/e2e/network/ingress.go

```diff
@@ -704,3 +647,84 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
 		})
 	})
 })
+
+func executePresharedCertTest(f *framework.Framework, jig *framework.IngressTestJig, staticIPName string) {
+	preSharedCertName := "test-pre-shared-cert"
+	By(fmt.Sprintf("Creating ssl certificate %q on GCE", preSharedCertName))
+	testHostname := "test.ingress.com"
+	cert, key, err := framework.GenerateRSACerts(testHostname, true)
+	Expect(err).NotTo(HaveOccurred())
+	gceCloud, err := framework.GetGCECloud()
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		// We would not be able to delete the cert until ingress controller
+		// cleans up the target proxy that references it.
+		By("Deleting ingress before deleting ssl certificate")
+		if jig.Ingress != nil {
+			jig.TryDeleteIngress()
+		}
+		By(fmt.Sprintf("Deleting ssl certificate %q on GCE", preSharedCertName))
+		err := wait.Poll(framework.LoadBalancerPollInterval, framework.LoadBalancerCleanupTimeout, func() (bool, error) {
+			if err := gceCloud.DeleteSslCertificate(preSharedCertName); err != nil && !errors.IsNotFound(err) {
+				framework.Logf("Failed to delete ssl certificate %q: %v. Retrying...", preSharedCertName, err)
+				return false, nil
+			}
+			return true, nil
+		})
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to delete ssl certificate %q: %v", preSharedCertName, err))
+	}()
+	_, err = gceCloud.CreateSslCertificate(&compute.SslCertificate{
+		Name:        preSharedCertName,
+		Certificate: string(cert),
+		PrivateKey:  string(key),
+		Description: "pre-shared cert for ingress testing",
+	})
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create ssl certificate %q: %v", preSharedCertName, err))
+
+	By("Creating an ingress referencing the pre-shared certificate")
+	// Create an ingress referencing this cert using pre-shared-cert annotation.
+	ingAnnotations := map[string]string{
+		framework.IngressPreSharedCertKey: preSharedCertName,
+		framework.IngressAllowHTTPKey:     "false",
```

Comment:
This doesnt seem to be testing this annotation?
Do we have a different test for it?

Reply:
Which test is it? We want that for kubemci as well :)
https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/142

Can you add that as a comment here (that we add this annotation to save resources, but is irrelevant to this test).

Reply:
That is ["should create ingress with given static-ip"](https://github.com/kubernetes/kubernetes/blob/bd6b5c80927a573e4db9f0d42746a7f06b219a4c/test/e2e/network/ingress.go#L124-L163). Sounds good to run it with kubemci, probably in follow-up :)

Will add  a comment.

Reply:
Done.